pubsub/api/PubsubTest.cs

```diff
@@ -0,0 +1,410 @@
+            subscriber)
+        {
+            // [START delete_subscription]
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, 
+                subscriptionId);
+            subscriber.DeleteSubscription(subscriptionName);
+            // [END delete_subscription]
+        }
+
+        public void DeleteTopic(string topicId, PublisherClient publisher)
+        {
+            // [START delete_topic]
+            string topicName = PublisherClient.FormatTopicName(_projectId, 
+                topicId);
+            publisher.DeleteTopic(topicName);
+            // [END delete_topic]
+        }
+
+        // [START retry]
+        /// <summary>
+        /// Retry the action when a Grpc.Core.RpcException is thrown.
+        /// </summary>
+        private T RetryRpc<T>(Func<T> action)
+        {
+            List<Grpc.Core.RpcException> exceptions = null;
+            var delayMs = _retryDelayMs;
+            for (int tryCount = 0; tryCount < _retryCount; ++tryCount)
+            {
+                try
+                {
+                    return action();
+                }
+                catch (Grpc.Core.RpcException e)
+                {
+                    if (exceptions == null)
+                        exceptions = new List<Grpc.Core.RpcException>();
+                    exceptions.Add(e);
+                }
+                System.Threading.Thread.Sleep(delayMs);
+                delayMs *= 2;  // Exponential back-off.
+            }
+            throw new AggregateException(exceptions);
+        }
+
+        private void RetryRpc(Action action)
+        {
+            RetryRpc(() => { action(); return 0; });
+        }
+
+        public int RpcRetry(string topicId, string subscriptionId,
+            PublisherClient publisher, SubscriberClient subscriber)
+        {
+            int tryCount = 0;
+            string topicName = PublisherClient.FormatTopicName(_projectId, topicId);
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, subscriptionId);
+            RetryRpc(() =>
+            {
+                //Create Topic only on second try
+                if (tryCount++ == 1)
+                {
+                    //Create Topic
+                    publisher.CreateTopic(topicName);
+                }
+
+                // Subscribe to Topic
+                // First try should fail
+                subscriber.CreateSubscription(subscriptionName, topicName, 
+                    pushConfig: null, ackDeadlineSeconds: 60);
+
+            });
+            return tryCount;
+        }
+        // [END retry]
+
+        private static bool IsEmptyResponse(PullResponse response)
+        {
+            foreach (var result in response.ReceivedMessages)
+                return false;
+            return true;
+        }
+
+        private static bool IsEmptyTopicsList(IEnumerable<Topic> topics)

```

Comment:
I think readers will be ok with seeing `topics.Count == 0`.  What do you think of dropping this function?